### PR TITLE
Add sql managed instance

### PR DESCRIPTION
**Pull request by jorgbaca**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by jorgbaca** [comment]
Hi @raych1 , have you had a chance to look at this PR?
Let me know if you need something from me. Thanks!

File: specification/azuredata/resource-manager/Microsoft.AzureData/preview/2019-07-24-preview/azuredata.json [hunk 1]
```diff
@@ -1094,22 +1633,12 @@
...
           {
-            "name": "postgresInstanceName",
-            "description": "Name of Postgres Instance",
+            "name": "dataControllerName",
             "in": "path",
             "required": true,
             "type": "string",
             "x-ms-parameter-location": "method"
           },
-          {
```
Comment (raych1) [inline specification/azuredata/resource-manager/Microsoft.AzureData/preview/2019-07-24-preview/azuredata.json:1104]: Removing required parameter is breaking change.

File: specification/azuredata/resource-manager/Microsoft.AzureData/preview/2019-07-24-preview/azuredata.json [hunk 2]
```diff
@@ -1863,6 +2386,261 @@
...
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/TrackedResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/DataControllerProperties",
+          "description": "null",
```
Comment (raych1) [inline specification/azuredata/resource-manager/Microsoft.AzureData/preview/2019-07-24-preview/azuredata.json:2415]: Can you add proper description or remove it?

Comment (jorgbaca) [reply specification/azuredata/resource-manager/Microsoft.AzureData/preview/2019-07-24-preview/azuredata.json]: Added a simple generic description.

File: specification/azuredata/resource-manager/Microsoft.AzureData/preview/2019-07-24-preview/azuredata.json (hunk 1 above)
Comment (jorgbaca) [reply specification/azuredata/resource-manager/Microsoft.AzureData/preview/2019-07-24-preview/azuredata.json:1104]: Hey @raych1 , the parameter was not removed. It's just the way GitHub shows the diff. Thanks.

**Comment by jorgbaca** [comment]
@raych1 @anthony-c-martin , I have this PR from some time ago that was never merged, I just solved a conflict from recent changes from master and all checks passed. Can you guys take a look at it? Let me know if you need something from me to get it merged. Thank you for your help.

**Review by chiragg4u — CHANGES_REQUESTED** [review]
Please take a look at the comments.

File: specification/azuredata/resource-manager/Microsoft.AzureData/preview/2019-07-24-preview/azuredata.json [hunk 3]
```diff
@@ -2151,6 +2704,261 @@
...
+        "Failed",
+        "PartialSuccess",
+        "Completed"
+      ],
+      "x-ms-enum": {
+        "name": "UsageUploadStatus",
+        "modelAsString": true
+      }
+    },
+    "HandshakeRequest": {
```
[the remark this answers is not in the document]
Comment (jorgbaca) [reply specification/azuredata/resource-manager/Microsoft.AzureData/preview/2019-07-24-preview/azuredata.json:2948]: Correct. There are some objects that still don't have properties. Most objects are just "shadow" objects that map to some objects on the customer's premises, but many functionality is still not defined/final yet.

File: specification/azuredata/resource-manager/Microsoft.AzureData/preview/2019-07-24-preview/examples/CreateOrUpdateDataController.json [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      "tags": {
+        "mytag": "myval"
+      },
+      "properties": {}
+    }
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "properties": {},
```
[the remark this answers is not in the document]
Comment (jorgbaca) [reply specification/azuredata/resource-manager/Microsoft.AzureData/preview/2019-07-24-preview/examples/CreateOrUpdateDataController.json:18]: For sure, this is for now just a copy of "hybridDataControllers" type which is being renamed to dataControllers. These objects are currently just used by a CLI tool that creates these "shadow" resources that map to onpremise objects. 

We plan on updating the examples and properties on these "shadow" resources as they work on premises progresses.

File: specification/azuredata/resource-manager/Microsoft.AzureData/preview/2019-07-24-preview/examples/CreateOrUpdateSqlManagedInstance.json [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+      "tags": {
+        "mytag": "myval"
+      },
+      "properties": {}
+    }
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "properties": {},
```
[the remark this answers is not in the document]
Comment (jorgbaca) [reply specification/azuredata/resource-manager/Microsoft.AzureData/preview/2019-07-24-preview/examples/CreateOrUpdateSqlManagedInstance.json:18]: This is just a copy of sqlInstances which is being renamed to sqlManagedInstances. We are working on what properties will be there and we'll add those once the work progresses. For now this is just a "shadow" resource that maps to some objects on premise.

File: specification/azuredata/resource-manager/Microsoft.AzureData/preview/2019-07-24-preview/examples/GetDataController.json [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  "parameters": {
+    "subscriptionId": "00000000-1111-2222-3333-444444444444",
+    "resourceGroupName": "testrg",
+    "dataControllerName": "testdataController",
+    "api-version": "2017-07-24"
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "properties": {},
```
[the remark this answers is not in the document]
Comment (jorgbaca) [reply specification/azuredata/resource-manager/Microsoft.AzureData/preview/2019-07-24-preview/examples/GetDataController.json:11]: Same here. Will update the examples as the work on the premise side progresses.

File: specification/azuredata/resource-manager/Microsoft.AzureData/preview/2019-07-24-preview/examples/ListSubscriptionDataController.json [hunk 7]
```diff
@@ -0,0 +1,34 @@
+{
+  "parameters": {
+    "subscriptionId": "00000000-1111-2222-3333-444444444444",
+    "api-version": "2017-07-24"
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "value": [
+          {
```
[the remark this answers is not in the document]
Comment (jorgbaca) [reply specification/azuredata/resource-manager/Microsoft.AzureData/preview/2019-07-24-preview/examples/ListSubscriptionDataController.json:10]: I'm not sure I understand what you mean here, can elaborate a little bit more?

File: specification/azuredata/resource-manager/Microsoft.AzureData/preview/2019-07-24-preview/azuredata.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (chiragg4u) [reply specification/azuredata/resource-manager/Microsoft.AzureData/preview/2019-07-24-preview/azuredata.json:2948]: If there are changes expected I would recommend you to publish to internal repo as per 
https://github.com/Azure/adx-documentation-pr/wiki/The-review-process

**Review by chiragg4u — APPROVED** [review]
Satisfied with the responses, approving.

**Review by raych1 — APPROVED** [review]

**Pull request merged**